language/move-binary-format/src/file_format.rs

```diff
@@ -1668,11 +1668,16 @@ impl Bytecode {
     /// Return the successor offsets of this bytecode instruction.
     pub fn get_successors(pc: CodeOffset, code: &[Bytecode]) -> Vec<CodeOffset> {
         assert!(
-            // The program counter could be added to at most twice and must remain
-            // within the bounds of the code.
-            pc <= u16::max_value() - 2 && (pc as usize) < code.len(),
+            // The program counter must remain within the bounds of the code
+            pc < u16::MAX && (pc as usize) < code.len(),
             "Program counter out of bounds"
         );
+
+        // No successor if pc is at the last instruction of the block. Return early to prevent overflow.
```

Comment:
Nit: comment is a bit confusing because this isn't actually checking whether the pc is at the last instruction in the block in the general case (only in the case where we have u16_max bytecode instructions).